aten/src/ATen/cuda/CUDABlas.cpp

```diff
@@ -576,27 +582,53 @@ void gemm<at::BFloat16>(CUDABLAS_GEMM_ARGTYPES(at::BFloat16)) {
       ldc,
       CUDA_R_32F,
       CUBLAS_GEMM_DEFAULT_TENSOR_OP));
-  TORCH_CUDABLAS_CHECK(cublasSetMathMode(handle, CUBLAS_DEFAULT_MATH));
 }
-#endif // !defined(USE_ROCM)
+#endif // defined(CUDA_VERSION) && CUDA_VERSION >= 11000
 
-#if !defined(USE_ROCM) && !defined(_MSC_VER)
+#if (defined(USE_ROCM) && ROCM_VERSION >= 50600) || (defined(CUDA_VERSION) && CUDA_VERSION >= 11000 && !defined(_MSC_VER))
 
 namespace {
+
+#if defined(USE_ROCM) && ROCM_VERSION >= 50600
+// hipify maps hipDataType -> hipDataType, but hipblaslt expects hipblasDatatype_t
+#define hipDataType hipblasDatatype_t
+// hipblaslt returns hipblasStatus_t, but hipify of TORCH_CUDABLAS_CHECK expects rocblas_status
+#define WORKAROUND_TORCH_CUDABLAS_CHECK(EXPR)                         \
+  do {                                                          \
+    hipblasStatus_t __err = EXPR;                               \
+    TORCH_CHECK(__err == HIPBLAS_STATUS_SUCCESS,                \
+                "HIP error: ",                                  \
+                at::cuda::blas::_hipblasGetErrorEnum(__err),    \
+                " when calling `" #EXPR "`");                   \
+  } while (0)
+// hipblaslt expexts hipblasDatatype_t, but it's a mess right now
+#define HIP_R_8I HIPBLAS_R_8I
+#define HIP_R_16F HIPBLAS_R_16F
+#define HIP_R_32I HIPBLAS_R_32I
+#define HIP_R_32F HIPBLAS_R_32F
+#define HIP_R_64F HIPBLAS_R_64F
+#define CUDA_R_16BF HIPBLAS_R_16B
+// hipify maps rocblas_status to rocblas_status, but we need hipblasStatus_t.
```

Comment:
typo: `rocblas_status` -> `cublasSt`

Reply:
hipification-time datatypes need to be macros, unfortunately